### PR TITLE
MEN-2285: Use FlushingWriter to automatically Sync() data to the bloc…

### PR DESCRIPTION
**Pull request by kacf**
…k device as we write.

Changelog: None
Signed-off-by: Jeremy Trimble <jeremy.trimble@twosixlabs.com>

### CODE REVIEW

**Comment by kacf** [comment]
This is the final piece of #342.

File: block_device.go [hunk 1]
```diff
@@ -34,11 +35,59 @@ type BlockDeviceGetSectorSizeFunc func(file *os.File) (int, error)
...
+	} else if fw.unflushedBytesWritten >= fw.FlushIntervalBytes {
+		err = fw.Sync()
+	}
+
+	return rv, err
+}
+
+func (fw *FlushingWriter) Sync() error {
+	err := fw.WF.Sync()
+	fw.unflushedBytesWritten = 0
```
Comment (lluiscampos) [inline block_device.go]: I think we should reset it to 0 only if `err != nil`

Comment (kacf) [reply block_device.go]: Disagree. If you actually get `err != nil` (which means you're likely in a world of trouble, but that's a different discussion), then not setting it to zero here will essentially cause a `Sync()` loop, where you constantly call it over and over as long as you write. There is likely nothing you can do if you get an error anyway, so might as well reset it.

**Comment by kacf** [comment]
Only a spurious error observed. Fixed comment in a followup commit and merged manually.

File: block_device.go (hunk 1 above)
Comment (lluiscampos) [reply block_device.go]: Makes sense. Thanks

**Pull request merged**